js/src/components/app-button/index.scss

```diff
@@ -10,12 +10,19 @@
 		}
 	}
 
+	// Add a gap between the SVG icon and text when the icon is on the right side.
+	// `:last-child` is to prevent this style apply on the loading <Spinner>.
+	&--icon-position-right.has-icon svg:last-child {
+		margin-left: 8px;
+	}
+
 	/**
-	 * Fix that the SVG icon gap to text should be on the left side.
-	 * @see https://github.com/WordPress/gutenberg/blob/%40wordpress/components%4012.0.8/packages/components/src/button/style.scss#L309-L311
+	 * Reset the padding when a button contains both text and icon.
+	 *
+	 * @see https://github.com/WordPress/gutenberg/blob/%40wordpress/components%4012.0.8/packages/components/src/button/style.scss#L292
+	 * @see https://github.com/WordPress/gutenberg/blob/%40wordpress/components%4012.0.8/packages/components/src/button/style.scss#L15
 	 */
-	&--fix-icon-position-right.has-icon.has-text svg {
-		margin-right: 0;
-		margin-left: 8px;
+	&--fix-padding.has-icon {
```

Comment:
I was thinking of just using `.app-button.has-icon` without the need of "--fix-padding", and I tested it out, it works for our case, but if we were to have a icon-only button without text, then the button would be in a slight rectangular shape instead of a square shape. 

💅  Instead of calling it "--fix-padding", how about "--icon-with-text"?

Reply:
I think it's ok to make its name more general instead of specifying the `padding` in it. Renamed by https://github.com/woocommerce/google-listings-and-ads/pull/963/commits/d98db6e187043be67448402bb7aa62289eee2999.